jane/doc/extensions/uniqueness/reference.md

```diff
@@ -0,0 +1,103 @@
+# Uniqueness Reference
+
+The goal of this document is to be a reasonably complete reference to the
+uniqueness mode. For a gentler introduction, see [the introduction](intro.md).
+It will grow over time as we add more features such as overwriting and
+borrowing.
+
+## More details on uniqueness checking
+
+### Mixing unique and aliased uses
+
+Continuing the example from [the introduction](intro.md), we can have both a
+unique and an aliased access to a value, as long as those occur in separate
+branches:
+
+```ocaml
+val free : t @ unique -> unit
+val store : t @ aliased -> unit
+
+let okay t =
+  if flip_coin ()
+  then free t
+  else store t; store t
+```
+
+This might be surprising coming from a language that uses ownership like Rust.
+In Rust, each value needs to have exactly one owner which is responsible for its
+deallocation. But in OCaml, we can rely on the GC for deallocation which makes
+it possible to give up ownership of the value in the second branch.
+
+### Unique uses of different fields
+
+The uniqueness analysis makes it possible to consume some fields of an
+allocation without losing access to the allocation or its other fields. For
+example, you might write:
+
+```ocaml
+let okay r =
+  free r.field1;
+  match r with
+  | { field2 } -> free field2
```

Comment:
Is it valid to completely omit field1 here? Or should this be `{ field2; _ }`

Reply:
It is valid to do this in the testsuite of the compiler. Would the internal codeformatter complain about this?

Reply:
Yeah, it emits the `missing-record-field-pattern` warning